### PR TITLE
Add to overload +operator for Spiller::Stats

### PR DESCRIPTION
**Pull request by xiaoxmeng**

### CODE REVIEW

File: velox/exec/Spiller.h [hunk 1]
```diff
@@ -218,6 +223,13 @@ class Spiller {
           spilledPartitions(_spilledPartitions) {}
 
     Stats() = default;
+
+    Stats& operator+=(const Stats& other) {
+      spilledBytes += other.spilledBytes;
+      spilledRows += other.spilledRows;
+      spilledPartitions = std::max(spilledPartitions, other.spilledPartitions);
```
Comment (mbasmanova) [inline velox/exec/Spiller.h:230]: Why is this a max and not a plus?

File: velox/exec/tests/SpillerTest.cpp [hunk 2]
```diff
@@ -701,7 +708,7 @@ class SpillerTest : public exec::test::RowContainerTestBase {
   std::unique_ptr<folly::IOThreadPoolExecutor> executor_;
   std::shared_ptr<TempDirectoryPath> tempDirPath_;
   std::shared_ptr<FileSystem> fs_;
-  RowTypePtr rowType_;
+  std::shared_ptr<const RowType> rowType_;
```
Comment (mbasmanova) [inline velox/exec/tests/SpillerTest.cpp:711]: use RowTypePtr

File: velox/exec/tests/SpillerTest.cpp [hunk 3]
```diff
@@ -869,7 +876,8 @@ TEST_P(NoHashJoinNoOrderBy, spillWithEmptyPartitions) {
 
 TEST_P(NoHashJoinNoOrderBy, spillWithNonSpillingPartitions) {
   // kOrderBy type which has only one partition, is irrelevant for this test.
-  RowTypePtr rowType = ROW({{"long_val", BIGINT()}, {"string_val", VARCHAR()}});
+  std::shared_ptr<const RowType> rowType =
```
Comment (mbasmanova) [inline velox/exec/tests/SpillerTest.cpp:879]: RowTypePtr or auto

File: velox/exec/tests/SpillerTest.cpp [hunk 4]
```diff
@@ -975,3 +983,38 @@ VELOX_INSTANTIATE_TEST_SUITE_P(
...
+
+  sumStats += stats;
+  EXPECT_EQ(stats.spilledRows, sumStats.spilledRows);
+  EXPECT_EQ(stats.spilledBytes, sumStats.spilledBytes);
+  EXPECT_EQ(stats.spilledPartitions, sumStats.spilledPartitions);
+
+  sumStats += stats;
+  EXPECT_EQ(2 * stats.spilledRows, sumStats.spilledRows);
+  EXPECT_EQ(2 * stats.spilledBytes, sumStats.spilledBytes);
+  // We take max of spill partitions.
```
Comment (mbasmanova) [inline velox/exec/tests/SpillerTest.cpp:1006]: Would you explain why?

**Review by mbasmanova — COMMENTED** [review]
@xiaoxmeng Looks good % one question.

File: velox/exec/Spiller.h (hunk 1 above)
Comment (xiaoxmeng) [reply velox/exec/Spiller.h:230]: Hmm, I were thinking when we sum up the spill stats from different operators and we kind of take the max as we spill the same set of partitions. It doesn't seem right to take a max also. I change to plus which is more meaningful. Thanks for pointing out.

Comment (mbasmanova) [reply velox/exec/Spiller.h:230]: After the change, should we expect spilledPartitions to be equal to total number of partitions times the number of drivers at the end of the query?

Comment (xiaoxmeng) [reply velox/exec/Spiller.h]: Yes

File: velox/exec/Spiller.h [hunk 5]
```diff
@@ -205,9 +210,9 @@ class Spiller {
 
   // Define the spiller stats.
   struct Stats {
-    uint64_t spilledBytes = 0;
-    uint64_t spilledRows = 0;
-    uint32_t spilledPartitions = 0;
+    uint64_t spilledBytes{0};
+    uint64_t spilledRows{0};
+    uint32_t spilledPartitions{0};
```
Comment (mbasmanova) [inline velox/exec/Spiller.h:215]: Would you document this value. Is it total number of partitions X number of drivers?

**Review by mbasmanova — APPROVED** [review]
@xiaoxmeng Thanks.

File: velox/exec/Spiller.h (hunk 5 above)
Comment (xiaoxmeng) [reply velox/exec/Spiller.h:217]: Commented. Thanks!

**Pull request closed without merging**